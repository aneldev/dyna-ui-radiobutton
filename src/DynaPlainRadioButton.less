.dyna-plain-radio-button{
	display: flex;
	align-items: center;
	cursor: pointer;

	.bullet-container{
		height: auto;
		font-size: 0;

		svg {
			stroke: currentColor;
			fill: currentColor;
		}
	}
	.text-container{
		height: auto;
		margin-left: 8px;
	}
}
