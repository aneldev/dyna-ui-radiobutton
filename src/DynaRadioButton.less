.dyna-radio-button {

	.dyna-plain-radio-button{
		position: absolute;
	}

	.width-helper{
		width: 20px !important;
	}

	input {
		position: absolute;
		left: -1000px; // hide it
	}

	&.dyna-ui-field-wrapper--size-LARGE{
		.dyna-ui-field-wrapper-control-container{
			position: relative;
			.dyna-plain-radio-button{
				top: 1px;
			}
		}
	}

	.dyna-ui-field-wrapper-container{
		min-width: 20px; // IE11 bug fix
	}


}
