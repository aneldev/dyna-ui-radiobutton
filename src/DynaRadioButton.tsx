import * as React from 'react';
import {Motion, MotionProps, spring, Style} from "react-motion";

const style: any = require('./DynaRadioButton.less');

export interface IDynaRadioButtonProps {
  checked?: boolean;
  caption?: string;
  size?: number;
  foregoundColor?: string;
  onChange?: (checked: boolean) => void;
}

export class DynaRadioButton extends React.Component<IDynaRadioButtonProps, any> {
  static defaultProps: IDynaRadioButtonProps = {
    foregoundColor: 'black',
    size: 20,
  };


  private handleClick():void{
    const {checked, onChange} = this.props;
    onChange(!checked);
  }

  public render(): JSX.Element {
    const {
      foregoundColor,
      size, caption,
      checked
    } = this.props;

    const strokeWidth: number = 1;
    const coreBulletWidth:number= checked ? (size/2) * 0.55 : 0;
	  let motionStyle: Style;
	  if (checked)
		  motionStyle = {bulletWidth: {stiffness: 300, damping: 10, val: coreBulletWidth, precision: 1}};
	  else
		  motionStyle = {bulletWidth: {stiffness: 300, damping: 26, val: coreBulletWidth, precision: 1}};

    return (
      <div className={style.container} onClick={this.handleClick.bind(this)}>
        <div className={style.bulletContainer}>
          <svg width={size} height={size} xmlns="http://www.w3.org/2000/svg">
            <g>
              <circle
                cx={size / 2}
                cy={size / 2}
                r={(size / 2) - strokeWidth}
                stroke={foregoundColor}
                strokeWidth={strokeWidth}
                fill="transparent"
              />
	            <Motion defaultStyle={{bulletWidth: 0}} style={motionStyle}>
		            {interpolatingStyle => (
			            <circle
				            cx={size / 2}
				            cy={size / 2}
				            r={interpolatingStyle.bulletWidth}
				            stroke={foregoundColor}
				            strokeWidth={strokeWidth}
				            fill="foregoundColor"
			            />
		            )}
	            </Motion>
            </g>
          </svg>
        </div>
        <div className={style.textContainer}>{caption}</div>
      </div>
    );
  }
}
