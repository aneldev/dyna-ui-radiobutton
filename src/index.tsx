export {
	DynaPlainRadioButton, IDynaPlainRadioButtonProps,
} from "./DynaPlainRadioButton";

export {
	DynaRadioButton, IDynaRadioButtonProps,
} from "./DynaRadioButton";

export {
	ESize, EStyle, EColor, EMode,
} from "dyna-ui-field-wrapper";
