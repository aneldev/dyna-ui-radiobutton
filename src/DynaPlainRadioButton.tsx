import * as React from 'react';
import {Motion, MotionProps, spring, Style} from "react-motion";

import "./DynaPlainRadioButton.less";

export interface IDynaPlainRadioButtonProps {
	checked?: boolean;
	label?: string;
	size?: number;
	foregoundColor?: string;
	onChange?: (checked: boolean) => void;
}

export class DynaPlainRadioButton extends React.Component<IDynaPlainRadioButtonProps> {
	static defaultProps: IDynaPlainRadioButtonProps = {
		checked: true,
		label: '',
		size: 20,
		foregoundColor: 'black',
		onChange: (checked: boolean) => undefined,
	};

	private handleClick(): void {
		const {checked, onChange} = this.props;
		onChange(!checked);
	}

	public render(): JSX.Element {
		const {
			size, label,
			checked
		} = this.props;

		const strokeWidth: number = 1;
		const coreBulletWidth: number = checked ? (size / 2) * 0.55 : 0;
		let motionStyle: Style;
		if (checked)
			motionStyle = {bulletWidth: {stiffness: 300, damping: 10, val: coreBulletWidth, precision: 1}};
		else
			motionStyle = {bulletWidth: {stiffness: 300, damping: 26, val: coreBulletWidth, precision: 1}};

		return (
			<div className="dyna-plain-radio-button" onClick={this.handleClick.bind(this)}>
				<div className="bullet-container">
					<svg width={size} height={size} xmlns="http://www.w3.org/2000/svg">
						<g>
							<circle
								cx={size / 2}
								cy={size / 2}
								r={(size / 2) - strokeWidth}
								strokeWidth={strokeWidth}
								fill="transparent"
							/>
							<Motion defaultStyle={{bulletWidth: 0}} style={motionStyle}>
								{(interpolatingStyle: Style) => (
									<circle
										cx={size / 2}
										cy={size / 2}
										r={interpolatingStyle.bulletWidth >= 0 ? Number(interpolatingStyle.bulletWidth) : 0}
										strokeWidth={strokeWidth}
									/>
								)}
							</Motion>
						</g>
					</svg>
				</div>
				<div className="text-container">{label}</div>
			</div>
		);
	}
}
