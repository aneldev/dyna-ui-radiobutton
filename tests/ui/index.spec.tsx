declare let global: any, jasmine: any, describe: any, clearTest: any, it: any, expect: any;

import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import * as React from 'react';
import * as enzyme from 'enzyme';

import {DynaPlainRadioButton} from '../../src';

describe('Home', () => {
  let wrapper;

  it('has expected content with deep render', () => {
    wrapper = enzyme.shallow(
      (
	      <DynaPlainRadioButton
		      checked={false}
		      onChange={() => console.log('I am clicked')}
	      />
      ),
      {}
    );

    expect(wrapper).toMatchSnapshot();
  });
});
