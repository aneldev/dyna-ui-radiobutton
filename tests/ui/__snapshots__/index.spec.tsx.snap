// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home has expected content with deep render 1`] = `
<div
  className="dyna-plain-radio-button"
  onClick={[Function]}
>
  <div
    className="bullet-container"
  >
    <svg
      height={20}
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <circle
          cx={10}
          cy={10}
          fill="transparent"
          r={9}
          strokeWidth={1}
        />
        <Motion
          defaultStyle={
            Object {
              "bulletWidth": 0,
            }
          }
          style={
            Object {
              "bulletWidth": Object {
                "damping": 26,
                "precision": 1,
                "stiffness": 300,
                "val": 0,
              },
            }
          }
        />
      </g>
    </svg>
  </div>
  <div
    className="text-container"
  />
</div>
`;
